<launch>
  <!--超时参数-->
  <param name="apoc_pkg/connect_timeout" type="double" value="27.7" />    <!--连接超时(单位：s)-->
  <param name="apoc_pkg/modeswitch_timeout" type="double" value="12.0" /> <!--模式切换超时(单位：s)-->
  <param name="apoc_pkg/armswitch_timeout" type="double" value="10.0" />  <!--arm/disarm超时(单位：s)-->
  <param name="apoc_pkg/fly_ab_timeout" type="double" value="30.0" />     <!--绝对飞行超时(单位：s)-->
  <param name="apoc_pkg/landing_timeout" type="double" value="7" />      <!--着陆超时(单位：s)--> 
  <param name="apoc_pkg/trace_timeout" type="double" value="10.0" />      <!--追踪飞行超时(单位：s)-->
  <param name="apoc_pkg/PID_FLIGHT_TIMEOUT" type="double" value="60.0" /> <!--PID飞行超时(单位：s) -->

  <!-- 容忍值参数 -->
  <param name="apoc_pkg/reach_tolerance_distance" type="double" value="0.03" />  <!--到达检测距离容忍值(单位：m)-->
  <param name="apoc_pkg/reach_tolerance_angle" type="double" value="0.1" />     <!--到达检测角度容忍值(单位：弧度？待确定)-->
  <param name="apoc_pkg/landing_tolerance" type="double" value="0.03" />        <!--着陆容忍值(单位：m)-->
  <param name="apoc_pkg/trace_tolerance" type="double" value="10.0" />          <!--追踪容忍值(单位：像素)-->

  <!-- 容忍值参数 -->
  <param name="apoc_pkg/flight_orien_x" type="double" value="0.0" />  <!--飞行姿态-->
  <param name="apoc_pkg/flight_orien_y" type="double" value="0.0" />  <!--四元数-->
  <param name="apoc_pkg/flight_orien_z" type="double" value="0.0" />  <!--=RPY(0,0,1)-->
  <param name="apoc_pkg/flight_orien_w" type="double" value="1.0" />  <!-- -->

  <param name="apoc_pkg/flight_alt" type="double" value="1.0" />  <!--到达检测距离容忍值(单位：m)-->


  <!--追踪参数-->
  <param name="apoc_pkg/trace_cam_ratio" type="double" value="0.00125" /><!--像素比例参数-->
  <param name="apoc_pkg/trace_target_center_x" type="double" value="320.0" /><!--像素目标中心x-->
  <param name="apoc_pkg/trace_target_center_y" type="double" value="240.0" /><!--像素目标中心y-->
  <!--  <param name="apoc_pkg/trace_tolerance" type="double" value="00.0" /> 在 容忍值参数 规定-->
  <!--  <param name="apoc_pkg/trace_timeout" type="double" value="60.0" />   在 超时参数   规定-->

  <!-- PID参数：X轴 -->
  <param name="apoc_pkg/PID_X_KP" type="double" value="1.0" />    <!-- 比例系数 -->
  <param name="apoc_pkg/PID_X_KI" type="double" value="0.0" />    <!-- 积分系数 -->
<<<<<<< Updated upstream
  <param name="apoc_pkg/PID_X_KD" type="double" value="0.0" />   <!-- 微分系数 -->
=======
  <param name="apoc_pkg/PID_X_KD" type="double" value="0.5" />   <!-- 微分系数 -->
>>>>>>> Stashed changes
  <param name="apoc_pkg/PID_X_OUT_MIN" type="double" value="-1.0" /> <!-- 输出最小值 -->
  <param name="apoc_pkg/PID_X_OUT_MAX" type="double" value="1.0" />  <!-- 输出最大值 -->
  <param name="apoc_pkg/PID_X_INT_MIN" type="double" value="-0.5" /> <!-- 积分最小值 -->
  <param name="apoc_pkg/PID_X_INT_MAX" type="double" value="0.5" />  <!-- 积分最大值 -->

  <!-- PID参数：Y轴 -->
  <param name="apoc_pkg/PID_Y_KP" type="double" value="1.0" />
  <param name="apoc_pkg/PID_Y_KI" type="double" value="0.0" />
<<<<<<< Updated upstream
  <param name="apoc_pkg/PID_Y_KD" type="double" value="0.0" />
=======
  <param name="apoc_pkg/PID_Y_KD" type="double" value="0.5" />
>>>>>>> Stashed changes
  <param name="apoc_pkg/PID_Y_OUT_MIN" type="double" value="-1.0" />
  <param name="apoc_pkg/PID_Y_OUT_MAX" type="double" value="1.0" />
  <param name="apoc_pkg/PID_Y_INT_MIN" type="double" value="-0.5" />
  <param name="apoc_pkg/PID_Y_INT_MAX" type="double" value="0.5" />

  <!-- PID参数：Z轴 -->
  <param name="apoc_pkg/PID_Z_KP" type="double" value="2.0" />
  <param name="apoc_pkg/PID_Z_KI" type="double" value="0.2" />
  <param name="apoc_pkg/PID_Z_KD" type="double" value="0.1" />
  <param name="apoc_pkg/PID_Z_OUT_MIN" type="double" value="-0.8" />
  <param name="apoc_pkg/PID_Z_OUT_MAX" type="double" value="2.8" />
  <param name="apoc_pkg/PID_Z_INT_MIN" type="double" value="-0.4" />
  <param name="apoc_pkg/PID_Z_INT_MAX" type="double" value="0.4" />

  <!-- PID参数：Yaw角 -->
  <param name="apoc_pkg/PID_YAW_KP" type="double" value="2.0" />
  <param name="apoc_pkg/PID_YAW_KI" type="double" value="0.05" />
  <param name="apoc_pkg/PID_YAW_KD" type="double" value="0.02" />
  <param name="apoc_pkg/PID_YAW_OUT_MIN" type="double" value="-0.6" />
  <param name="apoc_pkg/PID_YAW_OUT_MAX" type="double" value="0.6" />
  <param name="apoc_pkg/PID_YAW_INT_MIN" type="double" value="-0.3" />
  <param name="apoc_pkg/PID_YAW_INT_MAX" type="double" value="0.3" />

  <!-- PID控制频率 & 飞行超时 -->
  <param name="apoc_pkg/PID_CONTROL_RATE" type="int" value="50" />    <!-- 控制频率(Hz) -->
  <!--  <param name="apoc_pkg/PID_FLIGHT_TIMEOUT" type="double" value="60.0" />  在 超时参数   规定 -->
  
</launch>
